src/com/owncloud/android/files/FileMenuFilter.java

```diff
@@ -215,8 +215,12 @@ private void filter(List<Integer> toShow, List <Integer> toHide) {
                 (capability.getFilesSharingApiEnabled().isTrue() ||
                         capability.getFilesSharingApiEnabled().isUnknown()
                 );
+
+        boolean notAllowResharing = isSingleSelection() && anyFileSharedWithMe() &&
```

Comment:
No, it's not needed. This condition, indeed, it's being checked in the if just below and the share button will be hidden when is not a single selection.